pkg/maps/policymap/policymap.go

```diff
@@ -137,7 +154,7 @@ func OpenMap(path string) (*PolicyMap, bool, error) {
 	fd, isNewMap, err := bpf.OpenOrCreateMap(
 		path,
 		bpf.BPF_MAP_TYPE_HASH,
-		uint32(unsafe.Sizeof(uint32(0))),
+		uint32(unsafe.Sizeof(policyKey{})),
```

Comment:
The key size is not the same, what are the consequences for running instances in case we upgrade?

Reply:
Looks not great right now. I believe that after the new Cilium instance starts, it will attempt to regenerate the BPF (including PolicyMap) for existing endpoints, which will call down to OpenOrCreateMap(). If that latter function finds a map in the correct path, it won't validate that the map parameters are the same, so the existing map with a smaller valueSize will remain. Subsequently, when attempting to load the BPF program into the kernel, verification will fail due to access beyond the valueSize of the map. That will leave the container in a bad state.

I think that OpenOrCreateMap() should validate that all of the passed parameters are the same for the existing map, or delete it and attempt to create a new map to mitigate the above issue.

Reply:
Looks like there's already some validation in an alternative, OpenOrCreate() but not all of the maps are using this. Should be a minor extra patch.

Reply:
Commit https://github.com/cilium/cilium/pull/1599/commits/7c8f5024517cfb5a31f9251bd38e05a7513cf7c3 attempts to address this, would be keen to get some feedback on that.